WcaOnRails/spec/spec_helper.rb

```diff
@@ -84,4 +84,15 @@
   # test failures related to randomization by passing the same `--seed` value
   # as the one that triggered the failure.
   Kernel.srand config.seed
+
+  config.before(:suite) do
+    %w(Countries Continents Events Rounds Formats teams).each do |table|
+      # -- Uncomment to regenerate --
+      # ActiveRecord::Base.connection.execute("TRUNCATE #{table};")
+
+      if ActiveRecord::Base.connection.execute("SELECT COUNT(*) FROM #{table}").first == [0]
+        load "#{Rails.root}/db/seeds/#{table.underscore}.seeds.rb"
```

Comment:
I'd move your idea forward and create a file (ignored by git) somewhere inside spec directory called let's say `seeds_hash`. Then in the place we seed the data we concatenate the files contents into string and compute its hash. If it doesn't change then we can skip cleaning the tables. When it changes we write the new checksum to the `seeds_hash` and clean the tables to be reseeded. No need for database mess, what do you think?